documentation/src/main/asciidoc/userguide/chapters/caching/Caching.adoc

```diff
@@ -643,9 +643,140 @@ http://www.ehcache.org/documentation/2.8/integrations/hibernate#optional[Ehcache
+This option cannot be changed.
+
+*  All _entities and collections are configured to use a synchronous invalidation_ as clustering mode.
+This means that when an entity is updated, the updated cache will send a message to the other members of the cluster telling them that the entity has been modified.
+Upon receipt of this message, the other nodes will remove this data from their local cache, if it was stored there.
+This option can be changed to use replication by configuring entities or collections to use `replicated-entity` cache.
+How to change this option is explained in the <<caching-provider-infinispan-config, Configuration Properties>> section.
+
+*  All _entities and collections have initial state transfer disabled_ since there's no need for it.
+
+* Entities and collections are configured with the following eviction settings.
+You can change these settings on a per entity or collection basis or per individual entity or collection type.
+More information in the <<caching-provider-infinispan-config, Configuration Properties>> section below.
+ - Eviction wake up interval is 5 seconds.
+ - Max number of entries are 10,000
+ - Max idle time before expiration is 100 seconds
+ - Default eviction algorithm is LRU, least recently used.
+
+*  The query cache is configured so that _queries are only cached locally_ .
+Alternatively, you can configure query caching to use replication by selecting the `replicated-query` as query cache name.
+However, replication for query cache only makes sense if, and only if, all of this conditions are true:
+ - Performing the query is quite expensive.
+ - The same query is very likely to be repeatedly executed on different cluster nodes.
+ - The query is unlikely to be invalidated out of the cache (Note: Hibernate must aggressively invalidate query results from the cache any time any instance of one of the entity types targeted by the query. All such query results are invalidated, even if the change made to the specific entity instance would not have affected the query result)
+
+*  _query cache_ uses the _same eviction/expiration settings as for entities/collections_.
+
+*  _query cache has initial state transfer disabled_ . It is not recommended that this is enabled.
+
+*  The _timestamps cache is configured with asynchronous replication_ as clustering mode.
+Local or invalidated cluster modes are not allowed, since all cluster nodes must store all timestamps.
+As a result, _no eviction/expiration is allowed for timestamp caches either_.
+
+[IMPORTANT]
+====
+Asynchronous replication was selected as default for timestamps cache for performance reasons.
+A side effect of this choice is that when an entity/collection is updated, for a very brief period of time stale queries might be returned.
+It's important to note that due to how Infinispan deals with asynchronous replication, stale queries might be found even query is done right after an entity/collection update on same node.
```

Comment:
Could you log a JIRA to fix this? There are already some interceptors that do local update first, so it would be cool if you changed that (+ probably replacing CallInterceptor to make sure that timestamps increase monotonically).